From 6842bb3bd2a0b951420762a69c68691fc1f0d773 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Fri, 30 Dec 2016 19:27:47 +0100
Subject: ARM: dts: rockchip: add the MiQi board's fan definition

The MiQi board is sold with an enclosure in which a fan is connected
to the second LED output, and configured by default in "heartbeat"
mode so that it rotates slowly and increases when the CPU load
increases, ensuring appropriate cooling by default. This LED output
is called "Fan" in the original kernel and connected to GPIO18
(gpiochip 0, pin 18). Here we called it "miqi:green:fan" to stay
consistent with the kernel's naming conventions.

It's worth noting that without this patch the fan doesn't work at
all, risking to make the board overheat.

Fixes: 162718c (v4.7)
Cc: Heiko Stuebner <heiko@sntech.de>
Signed-off-by: Willy Tarreau <w@1wt.eu>
---
 arch/arm/boot/dts/rk3288-miqi.dts | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288-miqi.dts b/arch/arm/boot/dts/rk3288-miqi.dts
index 2f4c635..ef702db 100644
--- a/arch/arm/boot/dts/rk3288-miqi.dts
+++ b/arch/arm/boot/dts/rk3288-miqi.dts
@@ -67,6 +67,12 @@
 	leds {
 		compatible = "gpio-leds";
 
+		fan {
+			gpios = <&gpio0 18 GPIO_ACTIVE_HIGH>;
+			label = "miqi:green:fan";
+			linux,default-trigger = "heartbeat";
+		};
+
 		work {
 			gpios = <&gpio7 2 GPIO_ACTIVE_HIGH>;
 			label = "miqi:green:user";
-- 
2.8.0.rc2.1.gbe9624a

